### PR TITLE
:bug:Source Posthog: Fix Connector Date Formatting Error

### LINKED ISSUE #30166: Source Posthog: Connector Date Formatting Error
### Connector Name

source-posthog

### Connector Version

0.1.14

### What step the error happened?

During the sync

### Revelant information

Getting this error:
```
"internalMessage" : "No format in ['%Y-%m-%dT%H:%M:%S.%f%z'] matching 2023-07-08T13:00:28+00:00",
```

The existing state looks like this:
```
[
  {
    "streamDescriptor": {
      "name": "events"
    },
    "streamState": {
      "10874": {
        "timestamp": "2023-07-08T10:17:31.036000+00:00"
      },
      "17370": {
        "timestamp": "2023-07-08T13:00:28+00:00"
      }
    }
  }
]
```

So it seems like at some point the posthog timestamp formatting in regards to milliseconds changed which this connector does not like. 

### Relevant log output

```shell
2023-09-05 16:48:56 replication-orchestrator > failures: [ {
  "failureOrigin" : "source",
  "failureType" : "system_error",
  "internalMessage" : "No format in ['%Y-%m-%dT%H:%M:%S.%f%z'] matching 2023-07-08T13:00:28+00:00",
  "externalMessage" : "Something went wrong in the connector. See the logs for more details.",
  "metadata" : {
    "attemptNumber" : 1,
    "jobId" : 4461686,
    "from_trace_message" : true,
    "connector_command" : "read"
  },
  "stacktrace" : "Traceback (most recent call last):\n  File \"/airbyte/integration_code/main.py\", line 13, in <module>\n    launch(source, sys.argv[1:])\n  File \"/usr/local/lib/python3.9/site-packages/airbyte_cdk/entrypoint.py\", line 200, in launch\n    for message in source_entrypoint.run(parsed_args):\n  File \"/usr/local/lib/python3.9/site-packages/airbyte_cdk/entrypoint.py\", line 114, in run\n    yield from map(AirbyteEntrypoint.airbyte_message_to_string, self.read(source_spec, config, config_catalog, state))\n  File \"/usr/local/lib/python3.9/site-packages/airbyte_cdk/entrypoint.py\", line 156, in read\n    yield from self.source.read(self.logger, config, catalog, state)\n  File \"/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/declarative/manifest_declarative_source.py\", line 136, in read\n    yield from super().read(logger, config, catalog, state)\n  File \"/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py\", line 142, in read\n    raise e\n  File \"/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py\", line 122, in read\n    yield from self._read_stream(\n  File \"/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py\", line 197, in _read_stream\n    for record in record_iterator:\n  File \"/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py\", line 246, in _read_incremental\n    slices = stream_instance.stream_slices(\n  File \"/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/declarative/declarative_stream.py\", line 126, in stream_slices\n    return self.retriever.stream_slices()\n  File \"/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py\", line 348, in stream_slices\n    return self.stream_slicer.stream_slices()\n  File \"/airbyte/integration_code/source_posthog/components.py\", line 115, in stream_slices\n    project_datetime_slices = datetime_slicer.stream_slices()\n  File \"/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/declarative/incremental/datetime_based_cursor.py\", line 141, in stream_slices\n    start_datetime = self._calculate_earliest_possible_value(self._select_best_end_datetime())\n  File \"/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/declarative/incremental/datetime_based_cursor.py\", line 147, in _calculate_earliest_possible_value\n    cursor_datetime = self._calculate_cursor_datetime_from_state(self.get_stream_state())\n  File \"/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/declarative/incremental/datetime_based_cursor.py\", line 158, in _calculate_cursor_datetime_from_state\n    return self.parse_date(stream_state[self.cursor_field.eval(self.config)])\n  File \"/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/declarative/incremental/datetime_based_cursor.py\", line 196, in parse_date\n    raise ValueError(f\"No format in {self.cursor_datetime_formats} matching {date}\")\nValueError: No format in ['%Y-%m-%dT%H:%M:%S.%f%z'] matching 2023-07-08T13:00:28+00:00\n",</module>
  "timestamp" : 1693932509328
}, {
  "failureOrigin" : "source",
  "internalMessage" : "Source didn't exit properly - check the logs!",
  "externalMessage" : "Something went wrong within the source connector",
  "metadata" : {
    "attemptNumber" : 1,
    "jobId" : 4461686,
    "connector_command" : "read"
  },
  "stacktrace" : "io.airbyte.workers.internal.exception.SourceException: Source didn't exit properly - check the logs!\n\tat io.airbyte.workers.general.DefaultReplicationWorker.lambda$readFromSrcAndWriteToDstRunnable$5(DefaultReplicationWorker.java:326)\n\tat java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)\n\tat java.base/java.lang.Thread.run(Thread.java:1589)\nCaused by: io.airbyte.workers.exception.WorkerException: Source process exit with code 1. This warning is normal if the job was cancelled.\n\tat io.airbyte.workers.internal.DefaultAirbyteSource.close(DefaultAirbyteSource.java:145)\n\tat io.airbyte.workers.general.DefaultReplicationWorker.lambda$readFromSrcAndWriteToDstRunnable$5(DefaultReplicationWorker.java:324)\n\t... 4 more\n",
  "timestamp" : 1693932509769
}, {
  "failureOrigin" : "replication",
  "internalMessage" : "java.lang.IllegalStateException: Destination process is still alive, cannot retrieve exit value.",
  "externalMessage" : "Something went wrong during replication",
  "metadata" : {
    "attemptNumber" : 1,
    "jobId" : 4461686
  },
  "stacktrace" : "java.lang.RuntimeException: java.lang.IllegalStateException: Destination process is still alive, cannot retrieve exit value.\n\tat io.airbyte.workers.general.DefaultReplicationWorker.lambda$readFromDstRunnable$4(DefaultReplicationWorker.java:279)\n\tat java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)\n\tat java.base/java.lang.Thread.run(Thread.java:1589)\nCaused by: java.lang.IllegalStateException: Destination process is still alive, cannot retrieve exit value.\n\tat com.google.common.base.Preconditions.checkState(Preconditions.java:502)\n\tat io.airbyte.workers.internal.DefaultAirbyteDestination.getExitValue(DefaultAirbyteDestination.java:168)\n\tat io.airbyte.workers.general.DefaultReplicationWorker.lambda$readFromDstRunnable$4(DefaultReplicationWorker.java:263)\n\t... 4 more\n",
  "timestamp" : 1693932536011
} ]
```

### Contribute

- [X] Yes, I want to contribute

### PR DESCRIPTION
**Pull request by topefolorunso**
## What
Fix events stream cursor format resolves #30166

## Recommended reading order
1. `manifest.yaml`

## 🚨 User Impact 🚨
*Are there any breaking changes? What is the end result perceived by the user?*

*For connector PRs, use this section to explain which type of semantic versioning bump occurs as a result of the changes. Refer to our [Semantic Versioning for Connectors](https://docs.airbyte.com/contributing-to-airbyte/#semantic-versioning-for-connectors) guidelines for more information. **Breaking changes to connectors must be documented by an Airbyte engineer (PR author, or reviewer for community PRs) by using the [Breaking Change Release Playbook](https://docs.google.com/document/d/1VYQggHbL_PN0dDDu7rCyzBLGRtX-R3cpwXaY8QxEgzw/edit).***

*If there are breaking changes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.*

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-posthog/source_posthog/manifest.yaml [hunk 1]
```diff
@@ -116,7 +116,7 @@ definitions:
           end_datetime:
             datetime: "{{ now_utc().strftime('%Y-%m-%dT%H:%M:%S%z') }}"
             datetime_format: "%Y-%m-%dT%H:%M:%S%z"
-          datetime_format: "%Y-%m-%dT%H:%M:%S.%f%z"
+          datetime_format: "%Y-%m-%dT%H:%M:%S+00:00"
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-posthog/source_posthog/manifest.yaml:119]: look like braking change

**Review by lazebnyi — DISMISSED** [review]
need bump version and change log 

and follow with braking change flow or update to backward compatibility

File: airbyte-integrations/connectors/source-posthog/source_posthog/manifest.yaml (hunk 1 above)
Comment (topefolorunso) [reply airbyte-integrations/connectors/source-posthog/source_posthog/manifest.yaml:119]: No @lazebnyi , not a breaking change

**Review by marcosmarxm — APPROVED** [review]
I sent an update to correct the fix.

**Pull request merged**